### DISCUSSION TITLE
Can I run pg_restore during a migration?

### DISCUSSION THREAD

**Discussion by maresb** [sqlalchemy/alembic#851] [Usage Questions]
I'm using Alembic with a PostgreSQL database. The `pg_restore` command can be used to restore certain tables from a backup file. I would like to run `pg_restore` during a migration.

I tried this, and it did not work. The problem appears to be a deadlock, as the migration's connection locks out the connection from the `pg_restore` process.

Suppose I have a migration script whose only job is to run a `pg_restore` process. Is there any way to temporarily close the connection so that `pg_restore` and then subsequent scripts can run?

If I were to try to solve this myself, I would alternate between Alembic and `pg_restore`, using Alembic to run the migrations between backup scripts. But this feels like an inelegant hack, and I'm hoping for a better way. Thank you!

**Comment by zzzeek** [comment]
if this operation is in the midst of an actual migration, put the database connection into autocommit mode using [autocommit_block](https://alembic.sqlalchemy.org/en/latest/api/runtime.html?highlight=autocommit#alembic.runtime.migration.MigrationContext.autocommit_block).   all transactional state will be removed.
